src/services/motion/types.ts

```diff
@@ -0,0 +1,54 @@
+import pb from '../../gen/service/motion/v1/motion_pb';
+
+export type MotionConstraints = pb.Constraints.AsObject;
+
```

Comment:
Most of the code below will go away once we migrate to protobuf-es since that library generate protobuf structures that are essentially plain objects. for now we need to have custom functions to convert from objects -> protos.